test/e2e/run-all.js

```diff
@@ -66,7 +75,14 @@ async function main() {
     args.push('--retries', retries);
   }
 
-  for (const testPath of testPaths) {
+  // For running E2Es in parallel in CI
+  const currentChunkIndex = process.env.CIRCLE_NODE_INDEX ?? 0;
+  const totalChunks = process.env.CIRCLE_NODE_TOTAL ?? 1;
+  const chunkSize = testPaths.length / totalChunks;
```

Comment:
Hmm, won't this result in a float if the total isn't divisible by the total number of chunks? e.g. a total number of paths of 10 divided by a total chunk size of 3 would get 3.333... as the chunk size. Not ideal.

We'd want to round up, so that all paths were covered. So for 10 tests and 3 chunks, we'd want the chunk size to be 4.

```suggestion
  const chunkSize = Math.ceil(testPaths.length / totalChunks);
```

Reply:
<img width="364" alt="Screenshot 2022-11-08 at 14 32 55" src="https://user-images.githubusercontent.com/1561200/200579062-20e09f02-7429-41cd-98a6-70be2e8557b1.png">
All paths are covered regardless - but we can do `ceil` if you prefer 😅

Reply:
Well, that's an interesting result. Didn't realize `slice` would round down like that, and that it would effectively still work due to rounding. I was unable to find a case where this failed. Still I think `ceil` would be easier to read